uvicorn/middleware/debug.py

```diff
@@ -1,22 +1,33 @@
 import html
 import traceback
+from typing import Union
+
+from asgiref.typing import (
+    ASGI3Application,
+    ASGIReceiveCallable,
+    ASGISendCallable,
+    ASGISendEvent,
+    WWWScope,
+)
 
 
 class HTMLResponse:
-    def __init__(self, content, status_code):
+    def __init__(self, content: str, status_code: int):
         self.content = content
         self.status_code = status_code
 
-    async def __call__(self, scope, recieve, send):
+    async def __call__(
+        self, scope: WWWScope, receive: ASGIReceiveCallable, send: ASGISendCallable
+    ) -> None:
         await send(
-            {
+            {  # type: ignore
                 "type": "http.response.start",
                 "status": self.status_code,
                 "headers": [[b"content-type", b"text/html; charset=utf-8"]],
             }
```

Comment:
here we could avoid ignoring the type and do what we already did elsewhere with TypedDicts, something like
```
response_start: HTTPResponseStart = {"type": "http.response.start", ......}
await send(response_start)
```
I'm doing this on top of my head and cant remember if `HTTPResponseStart` exists but you see the spirit !

Reply:
It exists, but I had to change the list inside the headers, because it accepts `Iterable[Tuple[bytes, bytes]]`, but not the current `Iterable[List[bytes, bytes]]`, jfyk.

Reply:
> It exists, but I had to change the list inside the headers, because it accepts `Iterable[Tuple[bytes, bytes]]`, but not the current `Iterable[List[bytes, bytes]]`, jfyk.

that's a good mypy catch then :)